### DISCUSSION TITLE
How can I refer to the name of the connected device on an interface?

### DISCUSSION THREAD

**Discussion by whtkam** [netbox-community/netbox#12746] [Help Wanted!]
Is it possible to reference the connected endpoint device in a jinja2 template on an interfaces.

**Template:**
hostname {{ device.name }}
!
{% for interface in device.interfaces.all() %}
    interface {{ interface }}
    Connected-endpoint {{ interface.connected_endpoints.0}}
{% endfor %}

**Rendered Config:**
hostname MGMT-SW1
!
    interface GigabitEthernet0/0 (GigabitEthernet0/0)
    **Connected-endpoint eth0 (ethernet0)**

I only got the interface back but not the device name
What is the correct reference to get the device name of the connected device. In this example it should be "R2"

**REST-API - JSON of the interface**

      "connected_endpoints": [
        {
          "id": 50,
          "url": "https://192.168.0.1/api/dcim/interfaces/50/",
          "display": "eth0 (ethernet0)",
          "device": {
            "id": 8,
            "url": "https://192.168.0.1/api/dcim/devices/8/",
            "display": "R2",
            "name": "R2"
          },
          "name": "eth0",
          "cable": 5,
          "_occupied": true
        }
Thanks in advance!

**Comment by peterbaumert** [comment]
interface.connected_endpoints.0.device.name

**Reply by whtkam** [reply]
I already tried that but it didn't work

**Template**

hostname {{ device.name }}
!
{% for interface in device.interfaces.all() %}
interface {{ interface }}
Connected-endpoint {{ interface.connected_endpoints.0}}
Connected-endpoint {{ interface.connected_endpoints.0.device.name }}
{% endfor %}

**Rendered Config**

Traceback (most recent call last):
  File "/opt/netbox/netbox/dcim/views.py", line 2075, in get_extra_context
    rendered_config = config_template.render(context=context_data)
  File "/opt/netbox/netbox/extras/models/configs.py", line 252, in render
    output = template.render(**context)
  File "/opt/netbox/venv/lib/python3.8/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/opt/netbox/venv/lib/python3.8/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "<template>", line 6, in top-level template code
  File "/opt/netbox/venv/lib/python3.8/site-packages/jinja2/sandbox.py", line 326, in getattr
    value = getattr(obj, attribute)
jinja2.exceptions.UndefinedError: list object has no element 0

**Reply by peterbaumert** [reply]
That looks like one of your interfaces doesnt have something connected. You need to either check with if or use jinja default.

**Comment by whtkam** [comment]
That's it!  -> Thank you very very much!
